### DISCUSSION TITLE
Issue with port forwarding.

### DISCUSSION THREAD

**Discussion by pacificnm** [community/community#68253] [Codespaces]
### Select Topic Area

Product Feedback

### Body

None of the port forwarding is working for codespaces. Runing ReactJS and MeteorJS, tried adding ports manually and get the following error:

Unable to forward localhost:3000. Timed out performing tunnel_management_client_create_tunnel_port after 10000ms

**Comment by milindaShehan** [comment]
**Check Firewall Settings**
Ensure that there are no firewall rules or security software that might be blocking the ports you're trying to forward.

**Verify Configuration**
Double-check that you're using the correct port numbers for your ReactJS and MeteorJS applications. In your case, ReactJS commonly runs on port 3000 and MeteorJS on port 3000 or 4000 by default.

**Check if Ports are Already in Use**
It's possible that another process is already using the ports you're trying to forward. You can use a command like `netstat` or `lsof` to check which processes are listening on those ports.

**Restart Codespaces**
Try restarting the Codespace. Sometimes, there may be temporary issues that can be resolved by restarting.

**Use a Different Port**
If the default port (in this case, 3000) is giving you trouble, you could try using a different port for your application. For example, you could try using port 3001 or 4000.

**Check Codespaces Documentation**
Review the Codespaces documentation and community forums for any known issues or solutions related to port forwarding with ReactJS and MeteorJS.

**Review Codespaces Configuration**
Make sure that your Codespaces configuration is set up correctly. You might have missed a step or there could be an issue with your specific configuration.

**Update Codespaces**
If you're not using the latest version of Codespaces, consider updating to the latest version. Newer versions may have bug fixes or improvements related to port forwarding.

**Comment by nwogu** [comment]
Hello @pacificnm was this resolved? I'm facing similar issue, none of port forwarding is working for codespaces, Although I'm able to work around this using ngrok

**Reply by pacificnm** [reply]
I have not resolved it however i was able to use codspace using Visual Studio Code desktop version fine so I got around it, but kinda of annoying.

**Reply by nwogu** [reply]
Actually, mine is the opposite, Visual Studio Web, works well with port forwarding, but VSCode Desktop doesn't

**Comment by SaifurRR** [comment]
can you check your dns server?

**Reply by pacificnm** [reply]
Nothing wrong with DNS server or network, port forwarding works fine on another laptop on the same network. I think its something with the browser because on the same laptop that's having issue Visual Studio Code Desktop works fine. I am running same version of chrome with settings sync between the laptops, at a loss as to why one wont work.

**Reply by SaifurRR** [reply]
try removing cache from Chrome

**Comment by Spixmacaw** [comment]
Port forwarding issues can arise for various reasons, and troubleshooting them can require several steps. Here's a general guide to help you diagnose and resolve port forwarding problems:

**1. Verify Port Forwarding Configuration:**

   - Check that you have correctly configured port forwarding on your router or firewall. Ensure that the internal (private) IP address of the destination device is correctly set.
   - Confirm that the port number and protocol (TCP or UDP) you are forwarding are accurate.
   - Double-check that there are no typos or errors in your port forwarding settings.

**2. Firewall Settings:**

   - Ensure that there are no local software firewalls (e.g., Windows Firewall, iptables) on the destination device that might block incoming connections on the forwarded port.
   - Temporarily disable the firewall for testing purposes (not recommended for production systems) and see if the issue persists. If it works with the firewall disabled, you may need to create a rule to allow traffic on the forwarded port.

**3. Check External IP:**

   - Confirm that you are using the correct external (public) IP address when trying to access the forwarded port. Your public IP address may change if you have a dynamic IP from your ISP.

**4. Router/Modem Reboot:**

   - Sometimes, routers or modems may require a reboot to apply changes. Try restarting your router or modem to see if this resolves the issue.

**5. Port Availability:**

   - Ensure that the service/application you are trying to access on the destination device is running and actively listening on the forwarded port.

**6. ISP Restrictions:**

   - Some Internet Service Providers (ISPs) may block certain ports. Check if your ISP restricts the use of the port you are trying to forward.

**7. Double-NAT:**

   - If you have multiple routers or firewalls in your network, ensure that you are not dealing with a double-NAT situation. In double-NAT, port forwarding may need to be configured on both devices.

**8. Port Scanner Tools:**

   - Use online port scanner tools (e.g., "canyouseeme.org") to check if the forwarded port is open and reachable from the internet. This can help verify that your port forwarding is working as expected.

**9. Logging and Debugging:**

   - Check router or firewall logs for any relevant error messages or indications of blocked traffic.
   - On the destination device, examine the application/service logs to see if it's receiving incoming connections.

**10. IP Address Changes:**

   - Make sure the internal (private) IP address of the destination device hasn't changed. If it has, update your port forwarding rule accordingly.

**11. UPnP/Port Triggering:**

   - If you have Universal Plug and Play (UPnP) or port triggering enabled on your router, these features might interfere with manual port forwarding settings. Consider disabling them.

**12. Firmware/Software Updates:**

   - Ensure that your router/firewall's firmware is up-to-date. Manufacturers often release updates that can resolve known issues.

**13. Professional Help:**

   - If you've tried all of the above steps and still can't resolve the issue, consider seeking help from a networking professional or your ISP.

Remember to take appropriate security precautions when configuring port forwarding, as it can expose your devices to the internet. Always use strong passwords and consider using non-standard port numbers to enhance security.

**Comment by osortega** [comment]
@pacificnm 
I'm sorry you are experiencing this issues.
I took a look at your codespaces and it does seem that something in your browser could be causing the request to not go through.

Could you try with a couple of browsers? There could also be a VPN or local firewall blocking the request?
Could you also try maybe disabling browser extensions that could be intercepting the requests?

If this issue persists please reach out to GitHub support so that we can get more information from you like network HAR file and logs.

**Comment by AnimMouse** [comment]
Unblock this domain in your firewall: `tunnels.api.visualstudio.com`

**Comment by 01582** [comment]
same issue

**Comment by 01582** [comment]
since im on codespace so i just restarted my cluster
